src/storage/store/struct_column.cpp

```diff
@@ -74,10 +74,8 @@ void StructColumn::write(node_group_idx_t nodeGroupIdx, offset_t offsetInChunk,
     ValueVector* vectorToWriteFrom, uint32_t posInVectorToWriteFrom) {
     KU_ASSERT(vectorToWriteFrom->dataType.getPhysicalType() == PhysicalTypeID::STRUCT);
     if (vectorToWriteFrom->isNull(posInVectorToWriteFrom)) {
-        setNull(nodeGroupIdx, offsetInChunk);
         return;
     }
-    nullColumn->write(nodeGroupIdx, offsetInChunk, vectorToWriteFrom, posInVectorToWriteFrom);
```

Comment:
That was part of what I mentioned in the PR description: In-place updates on the null column may no longer always be possible. Though in practice, in-place updates for constant compressed data will do nothing but waste time doing an unnecessary write as there is no meaningful update that can be done to a constant compressed column (unless there is only one value, which isn't a case I'd included special handling for, but we'll still turn that into a new constant compressed chunk when doing the out of place update, even if it's a little slow compared to just updating the metadata with the new value).

> To get it working for nulls I had to set up out of place commits for the null column. This always happens when the parent column does an out of place commit (unnecessary, but it would take a lot of work to de-couple that I think), but we may now need to do an out-of-place commit for the null column when doing an in-place commit on its parent column. As such, **I've removed in-place updates to the null column when we do in-place updates to the parent column**, and have it otherwise doing an out-of-place update on the null column only if the main column does an in-place update to avoid duplicating work.

Reply:
make sense. thanks for the clarification.